packages/flutter/lib/src/material/menu_bar.dart

```diff
@@ -0,0 +1,3115 @@
+const double _kDefaultSubmenuIconSize = 24.0;
+
+// The default spacing between the the leading icon, label, trailing icon, and
+// shortcut label in a _MenuBarItemLabel.
+const double _kLabelItemDefaultSpacing = 18.0;
+
+// The minimum spacing between the the leading icon, label, trailing icon, and
+// shortcut label in a _MenuBarItemLabel.
+const double _kLabelItemMinSpacing = 4.0;
+
+// The minimum horizontal spacing on the outside of the top level menu.
+const double _kTopLevelMenuHorizontalMinPadding = 4.0;
+
+const Map<ShortcutActivator, Intent> _kMenuTraversalShortcuts = <ShortcutActivator, Intent>{
+  SingleActivator(LogicalKeyboardKey.enter): ActivateIntent(),
+  SingleActivator(LogicalKeyboardKey.numpadEnter): ActivateIntent(),
+  SingleActivator(LogicalKeyboardKey.space): ActivateIntent(),
+  SingleActivator(LogicalKeyboardKey.gameButtonA): ActivateIntent(),
+  SingleActivator(LogicalKeyboardKey.escape): DismissIntent(),
+  SingleActivator(LogicalKeyboardKey.tab): NextFocusIntent(),
+  SingleActivator(LogicalKeyboardKey.tab, shift: true): PreviousFocusIntent(),
+  SingleActivator(LogicalKeyboardKey.arrowDown): DirectionalFocusIntent(TraversalDirection.down),
+  SingleActivator(LogicalKeyboardKey.arrowUp): DirectionalFocusIntent(TraversalDirection.up),
+  SingleActivator(LogicalKeyboardKey.arrowLeft): DirectionalFocusIntent(TraversalDirection.left),
+  SingleActivator(LogicalKeyboardKey.arrowRight): DirectionalFocusIntent(TraversalDirection.right),
+};
+
+/// A mixin for describing cascading menu hierarchies that are part of
+/// a [MenuBar] or other cascading menu.
+///
+/// This class is abstract, and so can't be used directly. Typically subclasses
+/// like [MenuItem] and [MenuItemGroup] are used in practice.
+///
+/// See also:
+///
+///  * [MenuBar], a widget that renders menus in Flutter with a Material design
+///    style.
+///  * [PlatformMenuBar], a widget that renders menu items using platform APIs
+///    instead of Flutter.
+mixin MenuItem on Diagnosticable implements Widget {
+  /// A required label displayed on the entry for this item in the menu.
+  ///
+  /// This is rendered by default in a [Text] widget.
+  /// The label appearance can be overridden by using a [labelWidget] to render
+  /// a different widget in its place.
+  String get label;
+
+  /// An optional widget that will be displayed in place of the default [Text]
+  /// widget containing the [label].
+  Widget? get labelWidget => null;
+
+  /// The optional shortcut that selects this [MenuItem].
+  ///
+  /// This shortcut is only enabled when [onSelected] is set.
+  MenuSerializableShortcut? get shortcut => null;
+
+  /// Returns any child [MenuItem]s of this item.
+  ///
+  /// Returns an empty list if this type of menu item doesn't have
+  /// children.
+  List<MenuItem> get menus => const <MenuItem>[];
+
+  /// Returns all descendant [MenuItem]s of this item.
+  ///
+  /// Returns an empty list if this type of menu item doesn't have
+  /// descendants.
```

Comment:
As a menu newbie, it's not clear to me the difference between a child and a descendant. Consider adding additional comments to clarify.